lib/service/service.go

```diff
@@ -2160,18 +2154,20 @@ func (process *TeleportProcess) initSSH() error {
 		// Start BPF programs. This is blocking and if the BPF programs fail to
 		// load, the node will not start. If BPF is not enabled, this will simply
 		// return a NOP struct that can be used to discard BPF data.
-		ebpf, err = bpf.New(cfg.SSH.BPF)
+		ebpf, err := bpf.New(cfg.SSH.BPF)
 		if err != nil {
 			return trace.Wrap(err)
 		}
+		defer func() { warnOnErr(ebpf.Close(), log) }()
 
 		// Start access control programs. This is blocking and if the BPF programs fail to
 		// load, the node will not start. If access control is not enabled, this will simply
 		// return a NOP struct.
-		rm, err = restricted.New(cfg.SSH.RestrictedSession, conn.Client)
+		rm, err := restricted.New(cfg.SSH.RestrictedSession, conn.Client)
 		if err != nil {
 			return trace.Wrap(err)
 		}
+		// TODO: are we missing rm.Close()
```

Comment:
This, as well as `authClient` and `storagePresence` are never closed. Any idea if this was intentional or an oversight?

Reply:
`storagePresence` is a wrapper around process storage, which is closed after all services are done during shutdown, so we must not close that.

`authClient` will close itself after `process.ExitContext()` is done; we don't explicitly close any other cache as returned by `process.newAccessCache` and I'm not sure if we should start.

Reply:
Ok, what about `rm` - should it be closed?

cc @jakule - I think the restricted session manager is leaky either way. Could you take a look? (even if we do call `rm.Close()`, nothing appears to close `l.events.EventCh`, so the `auditEventLoop` never terminates.